src/tests/firmwarecontrol/Firmware_Upgrade_Upgrade_001.test.js

```diff
@@ -0,0 +1,80 @@
+import { firmwareUpgrade, upgradeStatus } from '../../commonMethods/firmwarecontrol'
+import { pluginActivate, pluginDeactivate } from '../../commonMethods/controller'
+import constants from '../../commonMethods/constants'
+
+let listener
+
+export default {
+  title: 'Firmware Upgrade - 001',
+  description: 'Check the Upgrade of the device to the specified firmware',
+  context: {
+    state: 'completed',
+  },
+  setup() {
+    return this.$sequence([
+      () => pluginDeactivate.call(this, constants.firmwareUpgradePlugin),
+      () => pluginActivate.call(this, constants.firmwareUpgradePlugin),
+      () => {
+        listener = this.$thunder.api.FirmwareControl.on(
+          'upgradeprogress',
+          data => {
+            this.$data.write('upgradeStatus', data.status)
+          },
+          e => {
+            this.$log('Error subscribing to upgradeprogress: ', e)
+          }
+        )
+        return true
+      },
+    ])
+  },
+  steps: [
+    {
+      description: 'Upgrade the firmware and check whether it is completed',
+      sleep: 5,
+      test() {
+        //TODO - Update the test to enter the 'name', 'location', 'type', 'progressinterval', 'hmac' through User prompt
+        return firmwareUpgrade.call(this, 'name', 'location', 'type', 'progressinterval', 'hmac')
```

Comment:
Do we want to merge this already or wait for an ability to fill this in with meaningful values?